dali/kernels/reduce/reduce_common.cuh

```diff
@@ -21,14 +21,27 @@
 namespace dali {
 namespace kernels {
 
+template <typename T>
+DALI_FORCEINLINE __device__ T shfl_down(T &t, int n) {
+  constexpr unsigned FULL_MASK = 0xffffffffu;
+  return __shfl_down_sync(FULL_MASK, t, n);
+}
+
+template <typename T, int N>
+DALI_FORCEINLINE __device__ vec<N, T> shfl_down(vec<N, T> &t, int n) {
+  constexpr unsigned FULL_MASK = 0xffffffffu;
+  vec<N, T> ret;
+  IMPL_VEC_ELEMENTWISE(ret[i] = __shfl_down_sync(FULL_MASK, t[i], n));
+  return ret;
```

Comment:
Doesn't it raise warnings of unreachable code?
IMPL_VEC_ELEMENTWISE already has a `vec<N, R> result`
```suggestion
  IMPL_VEC_ELEMENTWISE(__shfl_down_sync(FULL_MASK, t[i], n));
```